Source/REFACTORING/Collection/DataSource/BaseCollectionDataSource.swift

```diff
@@ -6,37 +6,23 @@
 //  Copyright © 2021 Александр Кравченков. All rights reserved.
 //
 
-import Foundation
-
-protocol CollectionDataSource: UICollectionViewDataSource, UICollectionViewDataSourcePrefetching {
-    var provider: CollectionGeneratorsProvider? { get set }
-    var prefetchPlugins: PluginCollection< BaseCollectionPlugin <PrefetchEvent>> { get set }
-    var collectionPlugins: PluginCollection< BaseCollectionPlugin <CollectionEvent>> { get set }
-}
-
-public protocol CollectionGeneratorsProvider: AnyObject {
-    var generators: [[CollectionCellGenerator]] { get set }
-    var sections: [CollectionHeaderGenerator] { get set }
-}
-
-extension BaseCollectionManager: CollectionGeneratorsProvider { }
-
+import UIKit
 
 // Base implementation for UICollectionViewDataSource protocol. Use it if NO special logic required.
-open class BaseCollectionDataSource: NSObject {
+open class BaseCollectionDataSource: NSObject, CollectionDataSource {
 
     // MARK: - Properties
 
-    weak var provider: CollectionGeneratorsProvider?
+    weak public var provider: CollectionGeneratorsProvider?
 
-    var prefetchPlugins = PluginCollection<BaseCollectionPlugin<PrefetchEvent>>()
-    var collectionPlugins = PluginCollection<BaseCollectionPlugin<CollectionEvent>>()
+    public var prefetchPlugins = PluginCollection<BaseCollectionPlugin<PrefetchEvent>>()
+    public var collectionPlugins = PluginCollection<BaseCollectionPlugin<CollectionEvent>>()
 
 }
 
 // MARK: - UICollectionViewDataSource
 
-extension BaseCollectionDataSource: CollectionDataSource {
+extension BaseCollectionDataSource: UICollectionViewDataSource {
```

Comment:
```suggestion
extension BaseCollectionDataSource {
```
Протокол `CollectionDataSource` уже подписан на `UICollectionViewDataSource`, тоже самое с `UICollectionViewDataSourcePrefetching`

Через suggestion удалить UICollectionViewDataSourcePrefetching не могу(